gcc6 failures with -Werror #pragma

Getting a compiler error when building 0.99.3 with gcc 6 only:

```
[   82s] arraylist.c:97:32: error: '-Werror' is not an option that controls warnings [-Werror=pragmas]
[   82s]  #pragma GCC diagnostic ignored "-Werror"
[   82s]                                 ^~~~~~~~~
[   82s] arraylist.c:110:32: error: '-Werror' is not an option that controls warnings [-Werror=pragmas]
[   82s]  #pragma GCC diagnostic ignored "-Werror"
[   82s]                                 ^~~~~~~~~
```

introduced in 8c49ee7490a41486db1c134cae4e5bf6e8599138:

```
#pragma GCC diagnostic push
#ifdef __clang__
#pragma GCC diagnostic ignored "-Wincompatible-pointer-types"
#else
#pragma GCC diagnostic ignored "-Werror"
#endif
```

--- a/arraylist.c
+++ b/arraylist.c
@@ -94,7 +94,9 @@ array_list_add(struct array_list *arr, void *data)
 #ifdef __clang__
 #pragma GCC diagnostic ignored "-Wincompatible-pointer-types"
 #else
+#if __GNUC__ < 6
 #pragma GCC diagnostic ignored "-Werror"
+#endif
 #endif
 void
 array_list_sort(struct array_list *arr, int(*sort_fn)(const void *, const void *))
@@ -107,7 +109,9 @@ array_list_sort(struct array_list *arr, int(*sort_fn)(const void *, const void *
 #ifdef __clang__
 #pragma GCC diagnostic ignored "-Wincompatible-pointer-types"
 #else
+#if __GNUC__ < 6
 #pragma GCC diagnostic ignored "-Werror"
+#endif
 #endif
 void* array_list_bsearch(const void **key, struct array_list *arr,
 		int (*sort_fn)(const void *, const void *))

